fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java

```diff
@@ -53,6 +53,16 @@ default List<String> listTableNames(String dbName) throws DdlException {
         return Lists.newArrayList();
     }
 
+    /**
+     * Return all partition names of the table.
+     * @param databaseName the name of the database
+     * @param tableName the name of the table
+     * @return a list of partition names
+     */
+    default List<String> listPartitionNames(String databaseName, String tableName) {
```

Comment:
why do we need this API

Reply:
this is required in prune partition by MetadataMgr.listPartitionNames